packages/server/graphql/mutations/helpers/handleCompletedStage.ts

```diff
@@ -16,7 +24,47 @@ const handleCompletedRetrospectiveStage = async (
   dataLoader: DataLoaderWorker
 ) => {
   if (stage.phaseType === REFLECT || stage.phaseType === GROUP) {
-    const data = await removeEmptyReflections(meeting)
+    const data: Record<string, any> = await removeEmptyReflections(meeting)
+
+    if (stage.phaseType === REFLECT) {
+      const r = await getRethink()
+      const now = new Date()
+      const reflectionGroups = await dataLoader
+        .get('retroReflectionGroupsByMeetingId')
+        .load(meeting.id)
+      const reflections = await r
+        .table('RetroReflection')
+        .getAll(meeting.id, {index: 'meetingId'})
+        .filter({isActive: true})
+        .orderBy('createdAt')
+        .run()
+      const {reflectionGroupMapping, autoGroupThreshold, nextThresh} = groupReflections(
+        reflections,
+        AUTO_GROUPING_THRESHOLD
+      )
+      console.log(`autoGroupThreshold = ${autoGroupThreshold}; nextThresh = ${nextThresh}`)
+
+      reflectionGroups.sort((groupA, groupB) => {
+        const newGroupIdA = reflectionGroupMapping[groupA.id]
+        const newGroupIdB = reflectionGroupMapping[groupB.id]
+        return newGroupIdA.localeCompare(newGroupIdB)
+      })
+
+      reflectionGroups.forEach(async (group, index) => {
+        group.sortOrder = index
+        await r
```

Comment:
don't use async/await in a `forEach`, it doesn't act how you think it acts!

instead, try `await Promise.all(reflectionGroups.map(() => return r.table(...))`